lib/pbench/cli/server/report.py

```diff
@@ -435,13 +452,15 @@ def summarize_dates(query: Query, options: dict[str, Any]):
         if date >= start and date < end:
             in_range += 1
 
-    click.echo(
-        f"  {in_range:,d} since {start:%Y-%m-%d %H:%M} until {end:%Y-%m-%d %H:%M}"
-    )
-    click.echo(f"    {this_year:,d} in year {year:%Y}")
-    click.echo(f"    {this_month:,d} in month {month:%B %Y}")
-    click.echo(f"    {this_week:,d} in week {week:%B %d} to {day:%B %d}")
-    click.echo(f"    {this_day:,d} on {day:%d %B %Y}")
+    click.echo(f"  {in_range:,d} from {start:%Y-%m-%d %H:%M} to {end:%Y-%m-%d %H:%M}")
+    if start <= year:
+        click.echo(f"    {this_year:,d} in year {year:%Y}")
+    if start <= month:
+        click.echo(f"    {this_month:,d} in month {month:%B %Y}")
+    if start <= week:
+        click.echo(f"    {this_week:,d} in week {week:%B %d} to {day:%B %d}")
+    if start <= day:
+        click.echo(f"    {this_day:,d} on {day:%d %B %Y}")
```

Comment:
This might get rid of the redundant outputs, which would be very good, but I question whether someone who actually specifies `--until` will be interested in these buckets.

That said, if `start` and `end` are in the same year, then isn't `start <= year` true and doesn't `this_year` have the same value as `in_range`?  So, shouldn't the condition be strictly less-than in each of these cases?

Reply:
"someone who actually specifies `--until`" is ... me. So, yeah, I find it interesting.

`year` is a `datetime.datetime` set to the beginning of the year; so on `--since 2024-01-02`, `start > year`, and we'll skip this.

I took a billionth or two of a second to debate internally whether to use `<` vs `<=`. Yes, if `--since` falls right at the beginning of a bucket, the total is redundant with that bucket -- it's debatable whether that's an issue. I was really thinking about weeks when I made that decision, the only bucket where it's not inherently obvious whether `--since` lines up with a bucket, and total the same as "this week" arguably has some significance. Whether that significance is significant is a different question, but I didn't overbelabor the point.